### PR TITLE
Benchmarking updates for semi-structured sparse training 

### PR DESCRIPTION
**Pull request by jcaip**
Summary:

This PR does the following:
- adds e2e ViT benchmarks for semi-structured sparse training
- adds nn.Linear microbenchmarks
- removes extra xformers benchmarking utils I copied over
- removes MLP block benchmarks
- updated README.md with new benchmarks + accuracy benchmarks

Given we have nn.Linear microbenchmarks and e2e benchmarks, I felt that
the MLP block benchmarks were unnecessary

As a sanity check, I ran the MLP benchmarks with the new benchmarking
suite and the old one, and got the same results:

NEW:
Screenshot 2024-06-18 at 6 16 37 PM

OLD:

|                        |   dense   |   w24    | 
|---------------------|-----------|----------|
| f16 (44160,1024,4096,1024)   |  11534.3  |  9204.7          |

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: torchao/sparsity/training/README.md [hunk 1]
```diff
@@ -42,19 +42,47 @@ swap_semi_sparse_linear_with_linear(model)
 
 ### Benchmarking
 
-If you want to see the expected speedups of applying runtime semi-structured sparsity for training, you can do so by modifying the existing benchmark code in to add your matmul shapes in:
-[benchmarks/benchamrk_semi_sparse.py](https://github.com/pytorch/ao/blob/main/benchmarks/benchmark_semi_sparse.py#L25)
+For ViT-L we see a **6% e2e speedup** on a single NVIDIA A100 across a single training (forwards + backwards) pass with torch.compile enabled and FP16 dtype:
```
Comment (msaroufim) [inline torchao/sparsity/training/README.md:45]: hmm while this is fine for for this tutorial not sure a top level ao provides 6% training speedups will land as strongly

**Review by msaroufim — APPROVED** [review]
Cool thank you! This is significantly clearer. I do want us to think a bit harder about the top line metric since 6% might not be super compelling to people not familiar with limitations of sparsity

**Comment by jcaip** [comment]
@msaroufim  We could compare to masking based approaches (which are slower than dense training) for a larger number, but I think it'd be a bit confusing since I'm assuming most users are coming with a dense model and not an existing sparse training script they want to accelerate.
